src/vos/vos_internal.h

```diff
@@ -810,6 +810,8 @@ struct vos_iter_info {
 	/* Reference to vos object, set in iop_tree_prepare. */
 	struct vos_object	*ii_obj;
 	d_iov_t			*ii_akey; /* conditional akey */
+	/** address range (RECX); 0:0 is is entire range (0:~0ULL) */
```

Comment:
what if they do something like 5:0 ?

It might be a simplification to just say if the length is 0, it covers whole range

As far as what you implemented below, it's actually if the length is 0, it means the rest of the address space from index...so it would actually be 

n:0 is remaining range (0:~0ULL)

I think it would be simpler if we just say length 0 means no range filter (or entire range)